tanner/emulators/template_injection.py

```diff
@@ -0,0 +1,51 @@
+import asyncio
+import logging
+import tornado
+
+from tanner.utils import patterns
+from tornado.template import Template
+from mako.template import Template
+from jinja2 import Environment
+
+Jinja2 = Environment()
+
+
+class TemplateInjection:
+    """
+        This emulator covers Jinja2, Mako and Tornado template engines.
+    """
+
+    def __init__(self, loop=None):
+        self._loop = loop if loop is not None else asyncio.get_event_loop()
+        self.logger = logging.getLogger('tanner.template_injection')
+
+    def get_injection_result(self, payload):
+
+        template_injection_result = None
+        base_template = """<html><head><title>Welcome %s !!</title></head></html>""" % payload
+
+        if patterns.TEMPLATE_INJECTION_MAKO.match(payload):
```

Comment:
This one is giving me 500 using `?a=<%= 7*7 %>`, look into it. I guess there is some URL decoding issue.

Reply:
For injections in `mako` you have to use something like this- `<% x=7*7 %>${x}`.

Reply:
I would like to know how you test the emulators. It looks like you don't test payloads using snare but directly execute the `get_injection_result` method.